Exposed loader in mirror_type_system to allow it to be overridden by cus...

shannah opened the pull request:
...tom implementation.

I have been using this modification for a while in the Mirah Netbeans module/ant task to make it possible to override the loader with my own.  It would be nice to add this to the core if it doesn't bother anybody.

baroquebobcat commented:
Could you tell me a little more about why you want to override the classloader? I'm not opposed to doing this, but it is exposing some of the type systems guts. We might be able to do it in a safer way.

shannah commented:
Hmm.. When I went in to get a sample of my usage, I found that I had already refactored my code so it doesn't look like I'm using this anymore.... let me look deeper as I may be missing something.
I was using this to add Java source file mirrors so that mirah and java source files could depend on each others during compile.  I changed it so that I just precompile java source files as stubs, then mirah can just use the binary java mirrors. 

Haven't looked at this in a while, so I may still need the behaviour...  Will get back to you.

ribrdb commented:
It does seem like we probably need something like this for ide integration, but I'm not sure if this is the best way to expose it.
